### PR TITLE
LPS-166629 AND LPS-166630 Add review/expiration date field to KB articles configuration

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://issues.liferay.com/browse/LPS-166629
https://issues.liferay.com/browse/LPS-166630

- LPS-166630 add the review and expiration dates as dateTogglerCheckboxLabel for the KBArticle
- LPS-166630 buildLang
- LPS-166630 save the values of expiration date and review date and obtain them on the field
- LPS-166630 the expiration date and the review date must be 1 year, and 9 moths on the future by default
- LPS-166630 reorder to match with service and do not create variables when it's not needed
- LPS-166630 put reviewDate and expirationDate on info panel
- LPS-166630 put errors on the top of the page to show them with others

This PR (under the feature.flag.LPS-165476 ) add the fields Review Date and the Expiration Date to the configuration panel of the KBArticle 

image

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp [hunk 1]
```diff
@@ -215,6 +215,32 @@ if (ListUtil.isEmpty(kbFolders) && ListUtil.isEmpty(kbArticles)) {
...
+						<dd class="sidebar-dd">
+							<c:choose>
+								<c:when test="<%= kbArticle.getReviewDate() != null %>">
+									<liferay-ui:message arguments="<%= new Object[] {dateFormatDateTime.format(kbArticle.getReviewDate()), HtmlUtil.escape(kbArticle.getUserName())} %>" key="x-by-x" translateArguments="<%= false %>" />
+								</c:when>
+								<c:otherwise>
+									<liferay-ui:message key="never-review" />
+								</c:otherwise>
+							</c:choose>
+						</dd>
```
Comment (marcogalluzzi) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp:243]: Is it OK to show this information if feature flag is not enabled?

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/info_panel.jsp:243]: yeah, totally, I forgot FF there.

**Comment by AliciaGarciaGarcia** [comment]
rebased and made changes

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:reevaluate:1377401_6275

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia @marcogalluzzi Is this ready to forward?

**Comment by marcogalluzzi** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1257401_5385

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
ci:reevaluate:1327401_6933

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1327401_6933

**Comment by Jeremy-Ch** [comment]
The above failures are not related to this PR, sent it to BChan: https://github.com/brianchandotcom/liferay-portal/pull/126393

**Pull request closed without merging**